src/main/java/ogc/rs/apiserver/util/Limits.java

```diff
@@ -78,6 +131,133 @@ private static Long parseDataUsageToBytes(String limit) {
+        if (bboxArray == null || bboxArray.isEmpty()) {
+            return null;
+        }
+
+        if (bboxArray.size() != 4) {
+            throw new OgcException(400, "Bad Request", INVALID_BBOX_FORMAT);
+        }
+
+        JsonArray validatedBbox = new JsonArray();
+        for (int i = 0; i < bboxArray.size(); i++) {
+            Object val = bboxArray.getValue(i);
+            if (!(val instanceof Number)) {
+                throw new OgcException(400, "Bad Request", ERR_BBOX_NON_NUMERIC);
+            }
+
+            double value = ((Number) val).doubleValue();
+            if (Double.isNaN(value) || Double.isInfinite(value)) {
+                throw new OgcException(400, "Bad Request", ERR_BBOX_NON_FINITE);
+            }
+
+            validatedBbox.add(value);
+        }
+
+        double minLon = validatedBbox.getDouble(0);
+        double minLat = validatedBbox.getDouble(1);
+        double maxLon = validatedBbox.getDouble(2);
+        double maxLat = validatedBbox.getDouble(3);
+
+        LOGGER.debug("minLon, minLat, maxLon, maxLat in token bbox: {} {} {} {}", minLon, minLat, maxLon, maxLat);
+
+        if (minLon < -180 || minLon > 180 || maxLon < -180 || maxLon > 180) {
+            throw new OgcException(400, "Bad Request", ERR_BBOX_LONGITUDE_RANGE);
+        }
+
+        if (minLat < -90 || minLat > 90 || maxLat < -90 || maxLat > 90) {
+            throw new OgcException(400, "Bad Request", ERR_BBOX_LATITUDE_RANGE);
+        }
+
+        if (minLon >= maxLon || minLat >= maxLat) {
+            throw new OgcException(400, "Bad Request", ERR_BBOX_MIN_MAX_ORDER);
+        }
+
+        return validatedBbox;
+    }
+
+    /**
+     * Parses and validates feature limits, returning as JsonObject for DataObject compatibility.
+     */
+    @GenIgnore
+    private static JsonObject parseFeatLimitToJsonObject(JsonObject featObject) {
+        if (featObject == null || featObject.isEmpty()) {
+            return null;
+        }
+
+        JsonObject validatedFeat = new JsonObject();
+
+        for (String collectionId : featObject.fieldNames()) {
+            // Validate that collectionId is a valid UUID
+            try {
+                UUID.fromString(collectionId);
+            } catch (IllegalArgumentException e) {
+                LOGGER.warn("Invalid collection ID format (must be UUID): {}", collectionId);
+                throw new OgcException(400, "Bad Request", "Invalid collection ID format (must be UUID) in the token");
+            }
+
+            JsonArray featureIds = featObject.getJsonArray(collectionId);
+            if (featureIds == null || featureIds.isEmpty()) {
+                LOGGER.warn("Invalid feature limit format for collection: {}", collectionId);
+                throw new OgcException(400, "Bad Request", "Invalid feature limit format for collection in the token");
+            }
+
+            JsonArray validatedFeatureIds = new JsonArray();
```

Comment:
Check for duplicate IDs as well

Reply:
If there are duplicate IDs, remove them from the array, no need to throw an error (I think).